src/Npgsql/PgPassFile.cs

```diff
@@ -74,11 +74,17 @@ class PgPassFile
             if (pgpassEnv != null)
                 return pgpassEnv;
 
-            // ReSharper disable AssignNullToNotNullAttribute
-            return PGUtil.IsWindows
-                ? Path.Combine(Environment.GetEnvironmentVariable("APPDATA"), "postgresql", "pgpass.conf")
-                : Path.Combine(Environment.GetEnvironmentVariable("HOME"), ".pgpass");
-            // ReSharper restore AssignNullToNotNullAttribute
+            if (PGUtil.IsWindows)
+            {
+                var appData = Environment.GetEnvironmentVariable("APPDATA");
```

Comment:
```suggestion
return Environment.GetEnvironmentVariable("APPDATA") is appData
    ? Path.Combine(appData, "postgresql", "pgpass.conf") : null;
```

Reply:
In that case the code can be changed to:

```csharp
return PGUtil.IsWindows
    ? Environment.GetEnvironmentVariable("APPDATA") is string appData ? Path.Combine(appData, "postgresql", "pgpass.conf") : null
    : Environment.GetEnvironmentVariable("HOME") is string home ? Path.Combine(home, "postgresql", "pgpass.conf") : null;
```

But this pattern introduces unnecessary casts. Don't think it makes code better.

Reply:
Since GetEnvironmentVariable returns string, I don't think that an "is string" pattern does anything more than check null... Your code proposal actually looks great to me.

Reply:
Checked, you're right.